include/igraph_adjlist.h

```diff
@@ -85,8 +85,7 @@ DECLDIR int igraph_inclist_init(const igraph_t *graph,
 DECLDIR int igraph_inclist_init_empty(igraph_inclist_t *il, igraph_integer_t n);
 DECLDIR void igraph_inclist_destroy(igraph_inclist_t *il);
 DECLDIR void igraph_inclist_clear(igraph_inclist_t *il);
-DECLDIR int igraph_inclist_remove_duplicate(const igraph_t *graph,
-        igraph_inclist_t *il);
+DECLDIR int igraph_inclist_remove_duplicate(igraph_inclist_t *il);
```

Comment:
I would not change this, because:

 - Based on our discussion about counting self-loops for eigenvector centrality and other cases, `adjlist_remove_duplicate` and `inclist_remove_duplicate` are not good solutions for what they do. Self-loop counting should be controlled when generating the adjlist/inclist. Therefore, we might soon deprecate this function. Thus, I would not create the impression that it is maintained.
 - The corresponding `adjlist_remove_duplicate` _must_ take a graph argument, consistency is worth maintaining
 - inclists, defined in terms of edge IDs, are not meaningful without a graph

Overall, I would prefer not to touch this function now (and get rid of it soon).

Reply:
Agreed.

Reply:
As said in other comments, let's do this in a separate PR. I have now added back in the `graph` argument.